### DISCUSSION TITLE
Will we get device.vendor with Reduced User Agent?

### DISCUSSION THREAD

**Discussion by dattanchu** [faisalman/ua-parser-js#667] [General]
Currently on Chrome with Reduced User Agent, even withClientHints() result, we are not getting the device.vendor field. For example, a Samsung phone would return with the model number SM-xxxx, but without the vendor field set to "SAMSUNG".

Do we plan to make this available for 2.0 stable?

**Comment by faisalman** [comment]
Currently client hints spec doesn't include vendor information, but the spec is not final, thus we still have no plan to accomodate the mentioned feature.

```js
// I haven't try this actually, but if we unreduce the user-agent first 
// and then parse as usual, maybe it will give different result?

import { UAParser } from 'ua-parser-js';
import { unfreezeUA } from 'ua-unfreeze-js';

unfreezeUA().then(newUA => console.log(UAParser(newUA).device.vendor));
```

**Comment by faisalman** [comment]
In the meantime you can also use [`getDeviceVendor()`](https://docs.uaparser.dev/api/submodules/helpers/get-device-vendor.html):

```js
const { device } = await UAParser().withClientHints();
if (!device.vendor && device.model) {
    device.vendor = getDeviceVendor(device.model);
}
```
